### DISCUSSION TITLE
Usage of custom elements in the WYSIWYG editor

### DISCUSSION THREAD

**Discussion by gitterbob** [GetPublii/Publii#1516] [Help]
`Hi,` 
I would like to integrate a Netlify Form into my site, but some parts of the HTML code being removed after saving the post locally.

Sample form:
```
<form name="contact" method="POST" data-netlify="true">
  <p>
    <label>Your Name: <input type="text" name="name" /></label>
  </p>
  <p>
    <label>Your Email: <input type="email" name="email" /></label>
  </p>
  <p>
    <label>Your Role: <select name="role[]" multiple>
      <option value="leader">Leader</option>
      <option value="follower">Follower</option>
    </select></label>
  </p>
  <p>
    <label>Message: <textarea name="message"></textarea></label>
  </p>
  <p>
    <button type="submit">Send</button>
  </p>
</form>
```

How to configure configure additional valid elements and custom elements correctly?

Thank you!

**Comment by bobmitro** [comment]
Take a look at your Site Settings.; ref: https://getpublii.com/docs/website-settings.html#editors

**Reply by candidexmedia** [reply]
To piggyback on Bob's answer: `*[*]` should do to include everything:

![image](https://github.com/GetPublii/Publii/assets/55474996/2e03526f-d0fe-4a58-b736-4be4d1316d4d)

Edit: it looks like all the `name=` attributes are still getting deleted. I'm running an older version of Publii, so hopefully it works better on newer versions.

Otherwise, try the Block Editor, and use the HTML block. I've had better luck with that in terms of preserving HTML code as is.

**Reply by gitterbob** [reply]
Thank you Bob and CM for your answers. The wildcard filter was a good idea, but I am experiencing the same as you. I use the recent version.

I will check now the Block editor.

**Reply by gitterbob** [reply]
Just a quick feedback: Block editor works perfectly, however WYSIWYG editor and additional valid elements might have some bug.

**Reply by peetersw** [reply]
Has anyone ever managed to get this to work? I am using the latest version of Publii.

**Reply by candidexmedia** [reply]
@peetersw I'll usually switch to the Block editor if it's too much trouble, tbh. I don't it's ever workee consistently for me.

What kind of tag or attribute are you trying to preserve?

**Reply by peetersw** [reply]
I'm trying to add 2 extra attributes "name" and "netlify-honeypot" to the form element. Every time when I save the page, these 2 attributes are deleted.

![Schermafbeelding 2025-06-06 180451](https://github.com/user-attachments/assets/2811cc8c-263e-4bf1-9895-501af8a2d8ad)

**Reply by gpsblues** [reply]
Hi,

In one of my blogs I implemented it simply using **Site settings > Editor > Additional valid elements > form[*]**

![{3E4893E2-B24B-4EE4-A809-8A0027FFD5FE}](https://github.com/user-attachments/assets/3bbb3f65-98be-44f2-a2b8-eaaf174a799a)


The HTML code in the WYSIWYG editor is as follows:

```html
<form action="/mail-inviata" method="POST" data-netlify="true">
    <p><label for="ctnome">Name</label></p>
    <input id="ctnome" name="ctnome" required type="text" placeholder="">
    <p><label for="ctemail">Email</label></p>
    <input id="ctemail" name="ctemail" required type="email" placeholder="">
    <input class="nascosto" name="per-robot" type="text">
    <p><label for="testo">Message</label></p>
    <textarea id="testo" name="testo" required rows="5" placeholder=""></textarea>
    <ul>
        <li>
            <input id="privacy" name="privacy" required type="checkbox" value="1">
            <label for="privacy">I have read and accept the <a href="#INTERNAL_LINK#/post/1">privacy policy</a></label>
        </li>
    </ul>
    <p><br><input class="btn--dark" type="submit" value="Send"></p>
</form>

**Reply by peetersw** [reply]
I've been able to get it working. However to me it seems like strange behaviour. If I use the attribute "name" everything will be removed every time when I save. But when I use "id" instead of "name" everything is kept. And the "id" attribute is recognized by Netlify for their forms.

**Reply by candidexmedia** [reply]
> In one of my blogs I implemented it simply using Site settings > Editor > Additional valid elements > form[*]
> I've been able to get it working.

I tried to add form[*], but I couldn't get the `name` nor the `netlify-honeypot` working on my end for Posts...

![image](https://github.com/user-attachments/assets/bc5a44bc-4c9a-43d8-a57c-f5ae6d7bffc0)

![Publii_forms](https://github.com/user-attachments/assets/da421f81-8613-4a4c-9ae5-3114ced52a70)

**Reply by peetersw** [reply]
use id instead of name, then it started working for me.

**Reply by candidexmedia** [reply]
> use id instead of name, then it started working for me.

What did you for the `netlify-honeypot`, though?

**Comment by peetersw** [comment]
![image](https://github.com/user-attachments/assets/cbc553ed-5f8d-40d8-ab15-3a3c243d0bf9)
